scripts/merge_tract_cat.py

```diff
@@ -184,33 +184,52 @@ def load_patch(butler_or_repo, tract, patch,
             print("  No good isPrimary entries for tract %d, patch %s" % (tract, patch))
         return ref_table
 
+    flux_field_names_per_schema_version = {
+        1: {'psf_flux': 'base_PsfFlux_flux', 'psf_flux_err': 'base_PsfFlux_fluxSigma',
+              'modelfit_flux': 'modelfit_CModel_flux', 'modelfit_flux_err': 'modelfit_CModel_fluxSigma'},
+        2: {'psf_flux': 'base_PsfFlux_flux', 'psf_flux_err': 'base_PsfFlux_fluxErr',
+              'modelfit_flux': 'modelfit_CModel_flux', 'modelfit_flux_err': 'modelfit_CModel_fluxErr'},
+        3: {'psf_flux': 'base_PsfFlux_instFlux', 'psf_flux_err': 'base_PsfFlux_instFluxErr',
+              'modelfit_flux': 'modelfit_CModel_instFlux', 'modelfit_flux_err': 'modelfit_CModel_instFluxErr'},
+    }
+
     merge_filter_cats = {}
     for filt in filters:
         this_data = tract_patch_data_id.copy()
         this_data['filter'] = filters[filt]
         try:
             cat = butler.get(datasetType='deepCoadd_forced_src',
-                             dataId=this_data).asAstropy()
+                             dataId=this_data)
         except NoResults as e:
             if verbose:
                 print(" ", e)
             continue
 
-        CoaddCalib = butler.get('deepCoadd_calexp_calib', this_data)
-        CoaddCalib.setThrowOnNegativeFlux(False)
+        if verbose:
+            print("AFW photometry catalog schema version: {}".format(cat.schema.VERSION))
+        flux_names = flux_field_names_per_schema_version[cat.schema.VERSION]
+
+        # Convert the AFW table to an AstroPy table
+        # because it's much easier to add column to an AstroPy table
+        # than it is to set up a new schema for an AFW table.
+        cat = cat.asAstropy()
 
-        mag, mag_err = CoaddCalib.getMagnitude(cat['base_PsfFlux_flux'], cat['base_PsfFlux_fluxSigma'])
+        calib = butler.get('deepCoadd_calexp_calib', this_data)
+        calib.setThrowOnNegativeFlux(False)
+
+        mag, mag_err = calib.getMagnitude(cat[flux_names['psf_flux']], cat[flux_names['psf_flux_err']])
```

Comment:
`'psf_fluxErr`

Reply:
Are you suggesting that in general the keys in the `flux_field_names` should be `psf_fluxErr` instead of `psf_flux_err`?

I understand the discussion and policy within the LSST Science Pipelines, but I find it easier to stick to a more traditional Python naming scheme when defining new alias names.

Reply:
The underscores have meaning in DM SourceCatalog Schemas: they separate things that aren't directly related. I'm not sure where we have it documented, but there is a method to it. `flux` and `fluxErr` are specific measurements, `psf` is the kind of that measurement.